scripts/apps/authoring/attachments/actions.ts

```diff
@@ -0,0 +1,89 @@
+import {gettext} from 'core/utils';
+
+export const INIT_ATTACHMENTS = 'INIT_ATTACHMENTS';
+export function initAttachments(item) {
+    return (dispatch, geState, {deployConfig, attachments}) => {
+        const payload = {
+            maxSize: deployConfig.getSync('attachments_max_size'),
+            maxFiles: deployConfig.getSync('attachments_max_files'),
+            files: [],
+        };
+
+        attachments.byItem(item)
+            .then((files) => {
+                payload.files = files;
+            })
+            .finally(() => {
+                dispatch({type: INIT_ATTACHMENTS, payload: payload});
+            });
+    };
+}
+
+export const ADD_ATTACHMENTS = 'ADD_ATTACHMENTS';
+export function selectFiles(files) {
+    return (dispatch, getState, {$scope, notify, superdesk}) => {
+        const state = getState();
+
+        if (Array.isArray(files) && files.length > 0 && !state.editor.isLocked) {
+            if (files.length + state.attachments.files.length > state.attachments.maxFiles) {
+                notify.error(gettext('Sorry, too many files selected.'));
+                return;
+            }
+
+            const bigFiles = files.filter((file) => file.size > state.attachments.maxSize);
+
+            if (bigFiles.length) {
+                notify.error(gettext('Sorry, but some files are too big.'));
+                return;
+            }
+
+            superdesk
+                .intent('upload', 'attachments', files)
```

Comment:
Could the intent thing be removed if redux is used? Which android developer added the intents here anyway? :D

Reply:
this opens a modal, for that I would probably go for another refactoring, will need to create support for that somehow